app/TuistApp/Sources/Views/SimulatorsView/SimulatorsView.swift

```diff
@@ -0,0 +1,88 @@
+import Path
+import SwiftUI
+import TuistCore
+import TuistServer
+import TuistSupport
+
+struct SimulatorsView: View, ErrorViewHandling {
+    @State var viewModel = SimulatorsViewModel()
+    @EnvironmentObject var errorHandling: ErrorHandling
+    @State var isExpanded = false
+
+    init(appDelegate: AppDelegate) {
+        // We can't rely on the SimulatorsView to be rendered before a deeplink is triggered.
+        // Instead, we listen to the deeplink URL through an `AppDelegate` callback
+        // that's eagerly set up in this `init` on startup.
+        let viewModel = viewModel
+        Task {
+            do {
+                try await viewModel.onAppear()
+            } catch {
+                ErrorHandling().handle(error: error)
+            }
+        }
+        appDelegate.onChangeOfURL = { url in
+            Task {
+                do {
+                    try await viewModel.onChangeOfURL(url)
+                } catch {
+                    ErrorHandling().handle(error: error)
+                }
+            }
+        }
```

Comment:
The `appDelegate.onChangeOfURL` API can lead to someone else overriding the value potentially causing some features to break. What if you use `Combine` in `AppDelegate` and turn it into an observer that multiple subscribers can subscribe to. 

I'd also consider using `[weak self]` since the app delegate is owned by the system, I don't think there's a need for views deep in the stack to hold a strong reference.

Reply:
There's no reference to self in this snippet at all. Neither we hold `appDelegate` strongly, so not exactly sure what you mean.

Definitely agree on using Combine `PassthroughSubject`, will update that 👍